### DISCUSSION TITLE
Static North Orientation Button for Android

### DISCUSSION THREAD

**Discussion by JHuberman** [OpenOrienteering/mapper#1819] [Ideas]
When field checking, it is quite useful to have the map oriented toward north, and the Android app has a button to do just that.
The problem I have encountered, is that when _Align Map With North_ is toggled on, the compass is adjusting the map every few seconds.  This makes it difficult to impossible to edit the map with _Align Map With North_ activated.  When I deactivate the button it reverts to North Up in relation to the tablet.

I would like to offer a suggestion to implement a _Static Align Map With North_ that is consistent with the way the _Move to My Location_ button is used.

When clicking the _Align Map With North_ , the map is rotated to align with north, but stays fixed until the button is toggled off at which point it returns to North Up in relation to the tablet.  A long click brings up a check box to _Keep Map Oriented to North_, and when checked the map would continuously reorient itself as it currently does as long as the feature is toggled on.

Since I am not a programer, to acomplish this, I made a harness with a rotating platform for the tablet, but a button click would be much more convenient.

![Tablet Harness](https://user-images.githubusercontent.com/17171179/103009587-feaf0200-4504-11eb-98b4-dc2c373b9f8d.jpg)
![Tablet Harness Tablet Attached](https://user-images.githubusercontent.com/17171179/103011296-ad544200-4507-11eb-85fc-3c97bff33073.jpg)

**Comment by mlerjen** [comment]
The autorotation actually pauses as soon as the pen approaches the tablet surface. I therefore use the pen as pausing device.

**Reply by dg0yt** [reply]
But this requires expensive proprietary pen/screen combinations. It does not work with standard touch screens and generic pens.

**Reply by mlerjen** [reply]
Ok. If we optionally could combine autorotation and autoposition in either permanent or on-request mode, as it is now for the position that would be great.

When I switched to work with autorotation it was an improvement to have the tools and toolbars always on fix positions (and not upside down). It also allowed me to work with sunshields.

**Reply by dg0yt** [reply]
In fact, I wonder if we should replace the existing auto-rotation with on-press rotation as suggested, perhaps even offering some averaging via press and hold. I suspect the autorotation (redrawing everything) based on the noisy compass sensor is a major driver for CPU power consumption.

**Reply by mlerjen** [reply]
The sunshield. Would otherwise be really hard to map above treeline.
![image](https://user-images.githubusercontent.com/15611653/103041446-bc260d80-4576-11eb-9dcd-42652618719e.png)

**Reply by lpechacek** [reply]
> In fact, I wonder if we should replace the existing auto-rotation with on-press rotation as suggested, 

I think, I would make use of this feature. I never adopted the autorotation due to the exact same reason as described in the initial post. Now, I rotate the tablet on my palm.

> perhaps even offering some averaging via press and hold.

Sounds like a good idea to me. Giving the magnetometer some time to settle after the button press and then locking the display would work. Alternatively, enhance autorotation to detect "rotation-less" periods and auto-lock display until a larger change in compass reading.

**Reply by dg0yt** [reply]
We may switch the existing compass button to the new behaviour, and offer a Sensors page setting to activate the old behaviour at least for an evaluation period.
Alternatively, we might offer a Sensors page setting to (additionaly) rotate the map on press of the Go-to-my-location button.

**Comment by dg0yt** [comment]
> Since I am not a programer, to acomplish this, I made a harness with a rotating platform for the tablet, but a button click would be much more convenient.

I use a Ski-O map holder.

**Comment by JHuberman** [comment]
I am really looking forward to the Move-to-my-location |
Align-map-with-north setting with no-automatic-updates.  I think I would
use it every time I stopped to add a feature.  Personally I don't think I
would use any of the automatic updates because I find it confusing to look
down and see that the map has moved.  Not rotating the tablet, it will be
nice to have the buttons stay in the same orientation and will certainly
save time.

Based on the comments in the thread I think the above is all that needs to
be implemented.

However, to add significant complexity, and to accommodate *all *the
suggestions in this thread the features could be implemented like this.

Using a long press to bring up check boxes to control the type of activity
for  Move-to-my-location  and Align-map-with-north controls offers
flexibility for the mapper in the field.  The settings page could allow
users to specify the seconds (or duration of long-press) for
averaging location and heading and/or distance moved to update location
and/or angle change to update bearing.

The settings page would offer:
Move-to-my-location settings:

GPS Averaging time for location  (set at 0 to average for long press
interval)

Distance traveled to trigger update (only relevant if
Keep-my-location-on-screen is checked.)


Align-map-with-north settings:

Averaging time for compass  (set at 0 to average for long press interval)

Degrees of change to trigger update  (only relevant if
Keep-my-orientation-current  is checked.)


The on-screen buttons would offer these choices on a long press:
Move-to-my-location

two radio buttons

North-up
Course-up

Check box

Keep-my-location-on-screen.

With the box checked the screen updates on the interval of the number of
seconds specified on the settings page.  With the box NOT checked the
screen updates only when pressed (after averaging for the set number of
seconds or the duration of the long press depending on the settings) and
then does not change until the button is pressed again or the set distance
is traveled.

The Align-map-with-north button would offer these choices:

two radio buttons

North-up
Course-up

Check box

Keep-my-orientation-current.

With the box checked the screen updates on the interval of the number of
seconds specified on the settings page.  With the box NOT checked the
screen updates only when pressed (after averaging for the set number of
seconds or the duration of the long press depending on the settings) and
then does not change until the button is pressed again or the heading of
the device changes the set number of degrees.

I think this covers all the suggestions mentioned in the thread and
maintains a consistency with the current user interface, but is probably
overly complex without demand for the additional control.


Joseph

================================
Joseph Huberman
904 Dorothea Drive
Raleigh, NC 27603

m. (919)610-1788

**Comment by JHuberman** [comment]
Kai,
I am checking to see if there is any activity on the StaticAlignWithNorth
option that we discussed in December?
I had an event on my first 1 sq/km map and it was well received and boosted
my enthusiasm and confidence.  I really appreciate all the work your team
has done to make this possible.
Joseph

================================
Joseph Huberman
904 Dorothea Drive
Raleigh, NC 27603

m. (919)610-1788

**Reply by dg0yt** [reply]
Joseph,
I don't think anyone is working on this feature. For my part of contributions, there is a still a small backlog of work-in-progess waiting to be finished, in particular to achieve a sufficient level of testing. 
Kai.

**Comment by JHuberman** [comment]
Thanks for the reply. I will be patient. My rotating platform works nicely
and my map is progressing well. I am hosting an event in two weeks on it.

Joseph Huberman
919-610-1788

**Comment by JHuberman** [comment]
Joseph Huberman
919-610-1788

**Comment by lpechacek** [comment]
@JHuberman, please re-post your suggestion as an issue ticket labeled "Enhancement:". I'll queue it in the mobile user interface enhancements. Thanks!
